components_library/memory/single_channel.py

```diff
@@ -104,3 +107,71 @@ def SingleChannelDDR3_2133(size: Optional[str] = None) -> AbstractMemorySystem:
     A single channel memory system using a single DDR3_2133_8x8 based DIMM
     """
     return SingleChannelMemory(DDR3_2133_8x8, size)
+
+
+def SingleChannelDDR4_2400(size: Optional[str] = None) -> AbstractMemorySystem:
+    """
+    A single channel memory system using a single DDR4_2400_8x8 based DIMM
+    """
+    return SingleChannelMemory(DDR4_2400_8x8, size)
+
+
+def SingleChannelLPDDR3_1600(
+    size: Optional[str] = None,
+) -> AbstractMemorySystem:
+    return SingleChannelMemory(LPDDR3_1600_1x32, size)
+
+
+def SingleChannelHBM(size: Optional[str] = None) -> AbstractMemorySystem:
+    return SingleChannelMemory(HBM_1000_4H_1x128, size)
+
+
+from .DS3.DS3Ctrl import DS3MemCtrl
+
+
+class DS3SingleChannel(AbstractMemorySystem):
+    def __init__(self, mem_type: str, size: Optional[str]):
```

Comment:
Please split all of this out into another python file. E.g., dramsim_3.py

So, you could then say
```python
from dramsim_3 import SingleChannelDDR3_1600
```
It would even be OK (preferable?) to have the same class names!